app/src/main/AndroidManifest.xml

```diff
@@ -7,6 +7,8 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
```

Comment:
No use of fine AFAIK, or is there?

Reply:
Here `LocationManager.NETWORK_PROVIDER` is used so fine location is required.

Reply:
According to https://developer.android.com/guide/topics/location/strategies.html